src/models/converters/transactions/details.rs

```diff
@@ -105,17 +108,19 @@ impl ModuleTransaction {
         &self,
         info_provider: &impl InfoProvider,
     ) -> ApiResult<TransactionDetails> {
+        let safe_transaction = &self.safe_transaction;
         Ok(TransactionDetails {
             executed_at: Some(self.execution_date.timestamp_millis()),
             tx_status: self.map_status(),
-            tx_info: self.to_transaction_info(info_provider).await,
+            tx_info: self.transaction_info(info_provider).await,
             tx_data: Some(TransactionData {
-                to: self.to.to_owned(),
-                value: self.value.to_owned(),
-                hex_data: self.data.to_owned(),
-                data_decoded: self.data_decoded.clone(),
-                operation: self.operation,
-                address_info_index: OptionFuture::from(self.data_decoded.as_ref().map(
+                to: safe_transaction.to.to_owned(),
+                value: safe_transaction.value.to_owned(),
+                hex_data: safe_transaction.data.to_owned(),
+                data_decoded: safe_transaction.data_decoded.clone(),
+                // TODO: check default value and if it is optional
+                operation: safe_transaction.operation.unwrap_or(Operation::CALL),
```

Comment:
I think we should resolve this `TODO` before merging. We could just talk about it in the internal services channel.

Reply:
Completely agree. Let's sync on this before merging!

Reply:
Fixed and rebased. Since optional was changed the object hash also changed. Fixed the test here: 02daef0

Reply:
Makes sense! We definitely need to do something about what we hash for this!